### DISCUSSION TITLE
Access to ref in Select component

### DISCUSSION THREAD

**Discussion by QWu4xYV** [chakra-ui/ark#1726] [Q&A] [answered]
I'm trying to use the React Select component together with react-hook-form, a common react form library. In simplest usage, it relies on refs to keep track of values/changes, but I can't figure out from the docs or the src how I would pass the ref to the underlying `<select>` element. Can someone give me a pointer? Thanks! :)

**Comment by cschroeter** [comment, marked as answer]
@QWu4xYV 

Does this answer your question? https://github.com/chakra-ui/ark/issues/1159#issuecomment-1630267020

**Reply by QWu4xYV** [reply]
Thank you @cschroeter ! That gives me what I need to integrate react-hook-form and Ark.

That said, it is slightly more verbose and doesn't answer the question of how I would get a ref if I need it for other reasons. Is there a reason that Ark doesn't expose refs?

**Reply by cschroeter** [reply]
Looking at the official docs how to [integrate with UI libraries](https://react-hook-form.com/get-started#IntegratingwithUIlibraries) it uses the same `Controller` approach that we also think is great.
